### PR TITLE
chore(ui-react-notifications): add InAppMessaging code

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Copy over InAppMessaging code to new package `ui-react-notifications`.  
InAppMessaging code in `ui-react` package will be removed in a future breaking change release.

#### Description of how you validated changes
Ran unit tests locally.
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by wlee221 — APPROVED** [review]

File: packages/react-notifications/src/components/InAppMessaging/CloseIconButton/CloseIconButton.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import { Button } from '@aws-amplify/ui-react';
+import { IconClose } from '@aws-amplify/ui-react/internal';
```
Comment (sreeramsama) [inline packages/react-notifications/src/components/InAppMessaging/CloseIconButton/CloseIconButton.tsx:4]: Is this okay?

Comment (wlee221) [reply packages/react-notifications/src/components/InAppMessaging/CloseIconButton/CloseIconButton.tsx:4]: IMO this is an expected usage, but someone else (cc @reesscot or @zchenwei) can verify

Comment (reesscot) [reply packages/react-notifications/src/components/InAppMessaging/CloseIconButton/CloseIconButton.tsx:4]: Yes, the icons are for internal use only. This means we can use them, but discourage them being used directly by customers.

**Comment by sreeramsama** [comment]
> Are there any differences between the ui-react InApp code and what is in this PR?

Mostly same; just fixed imports.

One change in test files mainly:
From
```
import { useBreakpointValue } from '../../../../hooks/useBreakpointValue';
jest.mock('../../../../hooks/useBreakpointValue');
```
to
```
import * as UiReactModule from '@aws-amplify/ui-react';
const useBreakpointValueSpy = jest.spyOn(UiReactModule, 'useBreakpointValue');
```

File: packages/react-notifications/src/components/InAppMessaging/FullScreenMessage/__tests__/FullScreenMessage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { render, screen } from '@testing-library/react';
+import * as UiReactModule from '@aws-amplify/ui-react';
+
+import { useMessageProps } from '../../hooks';
+import { BLOCK_CLASS } from '../constants';
+import { FullScreenMessage } from '../FullScreenMessage';
+import { FullScreenMessageProps } from '../types';
+import { BackdropProps } from '../../Backdrop/types';
+
+const useBreakpointValueSpy = jest.spyOn(UiReactModule, 'useBreakpointValue');
```
[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react-notifications/src/components/InAppMessaging/FullScreenMessage/__tests__/FullScreenMessage.test.tsx]: Done

File: packages/react-notifications/src/components/InAppMessaging/BannerMessage/__tests__/BannerMessage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import * as UiReactModule from '@aws-amplify/ui-react';
+
+import { useMessageProps } from '../../hooks';
+import { BLOCK_CLASS } from '../constants';
+import { BannerMessage } from '../BannerMessage';
+import { BannerMessageProps } from '../types';
+
+const useBreakpointValueSpy = jest.spyOn(UiReactModule, 'useBreakpointValue');
```
[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react-notifications/src/components/InAppMessaging/BannerMessage/__tests__/BannerMessage.test.tsx]: Done

**Pull request merged**